Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 485)
+++ CMakeLists.txt	(working copy)
@@ -49,6 +49,10 @@
   set_up_hermetic_build()
 endif()
 
+#if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    add_definitions(-DGTEST_USE_OWN_TR1_TUPLE=1)
+#endif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+
 # Instructs CMake to process Google Test's CMakeLists.txt and add its
 # targets to the current scope.  We are placing Google Test's binary
 # directory in a subdirectory of our own as VC compilation may break
