Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 692)
+++ CMakeLists.txt	(working copy)
@@ -51,6 +51,10 @@
 
 config_compiler_and_linker()  # Defined in internal_utils.cmake.
 
+#if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    add_definitions(-DGTEST_USE_OWN_TR1_TUPLE=1)
+#endif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+
 # Where Google Test's .h files can be found.
 include_directories(
   ${gtest_SOURCE_DIR}/include
