From f650765aab1a1bdba3cd4362c943747db4fa24b8 Mon Sep 17 00:00:00 2001
From: Christian Hoff <christian.hoff@bmw-carit.de>
Date: Thu, 19 Nov 2015 10:46:43 +0100
Subject: [PATCH] add fPIC compiler flag

---
 config.mk | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 config.mk

diff --git a/config.mk b/config.mk
new file mode 100644
index 0000000..0cbcb36
--- /dev/null
+++ b/config.mk
@@ -0,0 +1 @@
+CFLAGS += -fPIC
\ No newline at end of file
-- 
2.1.0

